### ISSUE TITLE
Run2.1i-y2wfd Processing Checklist

### ISSUE THREAD

**Issue by heather999** [LSSTDESC/ImageProcessingPipelines#88]
A checklist of items left to do as we prepare for the initial Run2.1i processing of the first 2 years of WFD data.

Review [Draft Run2.1i Processing Plan](https://github.com/LSSTDESC/ImageProcessingPipelines/wiki/DESC-DC2-Run2.1i-Preparation-Notes-%E2%80%94-DRAFT)
- [x] Set up meeting to review the document - Katrin, Jim, Heather, Dominique, Johann, Fabio
- [x] Adopt naming convention for processings i.e. Run2.1i-y2wfd, Run2.1i-y2, Run2.1i-y5
- [ ] Review naming conventions for the processing steps in the DM repo - Heather, Johann

Simulation Output and File Transfer to IN2P3
- [x] Update workflow to reduce/eliminate incomplete FITS file occurrences - Antonio
- [x] Complete Run2.1i visit DB - Antonio
- [x] Copy data to the appropriate area at NERSC - Antonio & workflow
- [x] Identify visits for eyeball checking - Jim & Alex
- [x] Process visits for eyeball checking - Heather
- [x] Validate simulation data and update visit DB accordingly - Heather & Tom
- [ ] File transfer to IN2P3 - Fabio

Reference Catalog
- [x] Update reference catalog for variable stars - Scott
- [x] Update ingestRefCat config to include new star variability column - Heather
- [x] ingest new reference catalog and provide tarball to share between IN2P3 and NERSC - Heather
- [x] Updated reference catalog to expand area - Scott
- [x] Ingest new reference catalog and provide tarball to share between IN2P3 and NERSC

Dome Flats
- [x] DM to provide updated DMstack w_2019_14 with DM-18703 fix for generating flats - DM in consultation with Jim
- [x] Generate Run2.1i flats, either with xtalk turned off or using the above weekly - Heather
- [x] Provide full 2.1i CALIB products tarball to IN2P3 including sky table. - Heather

BF kernels
- [x] Rereun BF kernel generation on test subset of BF flats with run number fix - Tom
- [x] Provide updated BF flats when DM-18703 is available in w_2019_14 - Heather
- [x]  Assess resulting test BF kernels - Jim
- [x]  Assess the BF kernels generated using the two available algorithms to determine if the kernels are adequate and choose one set of kernels for use in Run2.1i - Jim, Michael, Bob, Rachel
- [x] Deliver BF kernel data to IN2P3 - Heather & Fabio

Review Run1.2i SRS workflow versus workflow rewrite
- [x]  Is the run1.2.i workflow adequate for ingest->calexp->coadd->multiband? - Johann
- [x]  If not, assess work necessary to complete updated workflow - Johann

Work needed a minima on the 1.2i workflow
 - [ ]  housekeeping information put in place and tested for the 1.2i workflow units - Johann
 - [ ]  paths to execution ancillaries tidied up - Johann

Possible useful developments
- [ ]  Modify the existing pipelines so that they integrate better the new tasks : separate "per tract" tasks and "per subset of patch" tasks in coadd pipeline, anticipate the possibility that coaddDriver fail on full tract processing, etc...

Sky Correction
- [x] Provide list of incomplete fits files in initial Run2.1i production - Heather
- [X]  Fix Run1.2i CALIB to include sky table - Dominique
- [x] Identify a couple of visits or two from Run2.1i with dark halos in the calexps and source halos around stars - Jim
- [x]  Run Sky Correction Workflow on these id'd visits - Heather
- [x] Process Run2.1i batch 1 with and without sky correction through coadds - Johann
- [x]  Assess outputs to determine the utility of sky correction inclusion in the DM processing - ?
- [x]  If deemed ok, add sky correction to the SRS workflow - Johann

Working through Astrometry Problems during calexp generation
- [x] Work with DM to sort out MatchPessimisticB failures in SFD - Dominique & Bob
[DM-19645](https://jira.lsstcorp.org/browse/DM-19645)

Forced Photometry
- [x] Add forced photometry steps to the SRS workflow - Johann
- [x] Run Forced Photometry on Run1.2i data - Johann
- [x] Deliver outputs of Run1.2i processing to NERSC/data access group - Fabio

Metacal
- [x] Test running metacal on Run1.2i data, using the updated conda environment available since w_2019_13 - Bob, Johann
- [x] Based on our ability to run metacal, close [RFC-583](https://jira.lsstcorp.org/browse/RFC-583) - Heather
- [x] Work through any resulting run time errors and log messages with Erin - Bob, Johann
- [ ] Assess metacal output - Bob & WL Group
- [x] Pending further investigation, add metacal to the SRS workflow - Johann

NERSC preparations
- [ ] Clean up duplicate data on projecta - Heather
- [ ] Back up Run1.2p/i data to HPSS to free projecta space - Heather
- [ ] Back up Run2.1i output to HPSS as we go - Heather

DM and obs_lsst updates
- [x] [Update ingestCalib.py](https://github.com/lsst/obs_lsst/pull/95) to avoid omitting sky table - Heather
- [x] Confirm [DM-18643](https://jira.lsstcorp.org/browse/DM-18643) and required config updates are included in the weekly (w_2019_15?) to be used in processing - Heather
- [x] Watch https://github.com/lsst/obs_lsst/pull/98/files

**Comment by heather999** [comment]
@johannct I see the two checkmarks beside 
- [X] Is the run1.2.i workflow adequate for ingest->calexp->coadd->multiband? - Johann
- [X] If not, assess work necessary to complete updated workflow - Johann
Can you summarize?  Is the Run1.2i workflow adequate?  And if not, is there a time estimate for completing an updated workflow?

**Comment by heather999** [comment]
@johannct  I'm getting ready to close this issue, but I wanted to check on a few items.  Can you indicate if these are done, require a new issue to be opened so we continue to follow it, or just say we're not going do this right now?

Work needed a minima on the 1.2i workflow

- [ ] housekeeping information put in place and tested for the 1.2i workflow units - Johann
- [ ]  paths to execution ancillaries tidied up - Johann

Possible useful developments

- [ ] Modify the existing pipelines so that they integrate better the new tasks : separate "per tract" tasks and "per subset of patch" tasks in coadd pipeline, anticipate the possibility that coaddDriver fail on full tract processing, etc...

**Comment by johannct** [comment]
the housekeeping is still not great, but I decided to not touch to anything anymore, unless it is a bug fix. The next round will wait for gen3 and even perhaps the next workflow manager. So you can close.

**Comment by heather999** [comment]
We are well on our way with Run2.1i processing, closing
